src/main/java/com/bruincreates/server/service/AccountService.java

```diff
@@ -118,4 +120,66 @@ public void resetPassword(PasswordResetRequest request) throws BadRequestExcepti
         user.setPassword(passwordEncoder.encode(password));
         userMapper.updateByExampleSelective(user, userExample);
     }
+
+    public void updateAccount(AccountUpdateRequest request) throws BadRequestException {
+
+        //1. Validate new username
+        if(request.getNewUsername() != null){
+
+            //invalid new username
+            if(request.getNewUsername().equals("") || request.getNewUsername().contains(" ")) {
+                throw new BadRequestException("invalid new username");
+            }
+
+            //new username is already taken
+            User alreadyTaken = findUserByUsername(request.getNewUsername());
+            if(alreadyTaken != null) {
+                throw new BadRequestException("username already taken");
+            }
+
+        }
+
+        //2. Validate new email
+        if(request.getNewEmail() != null){
+
+            //check if the new email is valid
+            boolean validEmail = EmailValidator.getInstance().isValid(request.getNewEmail());
+            if (!validEmail) {
+                throw new BadRequestException("wrong email format");
+            }
+
+            //new email is already taken
+            User alreadyTaken = findUserByEmail(request.getNewEmail());
+            if(alreadyTaken != null) {
+                throw new BadRequestException("email already taken");
+            }
+
+        }
+
+        //3. Validate new profile name
+        if(request.getNewProfileName() != null){
+            //empty new profile name
+            if(request.getNewProfileName().equals("")) {
+                throw new BadRequestException("invalid new profile name");
+            }
+        }
+
+        //4. Now, after validating, update it all into the database
+        UserExample userExample = new UserExample();
+        userExample.createCriteria().andUsernameEqualTo(UserUtil.getRuntimeUser().getUsername());
```

Comment:
You are supposed to get username in the controller instead of service. Could take a reference from product controller